elastic/eui
# [EuiDataGrid] Various row height fixes

## Summary

- addreses https://github.com/elastic/eui/issues/7780#issuecomment-2183179080
- closes https://github.com/elastic/eui/issues/7897
- closes https://github.com/elastic/eui/issues/7811

This PR contains several fixes as well as storybook/VRT test updates. I strongly recommend [following along by commit](https://github.com/elastic/eui/pull/8025/commits), as I continued to update VRT screenshots as I fixed bugs, which is helpful in seeing the visual diffs.

## QA

- Go to https://eui.elastic.co/pr_8025/storybook/?path=/story/tabular-content-euidatagrid-rowheightsoptions-prop--custom-line-height
- Change the line height from 40px to, e.g. `5`
- [x] Confirm row heights update accordingly
- Click the display settings selector in the datagrid toolbar and switch the `row height` to `Custom` (2 lines)
- [x] Confirm that the control column checkbox and action buttons remain aligned with the first line of text, and do not have alignment issues/move when toggled between `Single` and `Custom`
- Go to https://eui.elastic.co/pr_8025/storybook/?path=/story/tabular-content-euidatagrid-gridstyle-prop--playground
- [x] Toggle the various `fontSize` and `cellPadding` option permutations and confirm that row height(s) update accordingly/as expected
- Go to https://eui.elastic.co/pr_8025/storybook/?path=/story/tabular-content-euidatagrid-rowheightsoptions-prop--custom-row-heights
- [x] Confirm the last row is truncated correctly at 2 lines, and its text is not incorrectly clipped
- Click the display settings selector in the datagrid toolbar and switch the density to `Compact` and the row height to `Custom` with `1` line per row
- [x] Confirm that the Version badges are visible until the end of the cell and are not cut off by the compact padding, i.e. <img width="100" alt="" src="https://github.com/user-attachments/assets/ee00a70a-4224-4c8b-9cf9-9d740e987cd0"> 


### General checklist

- Browser QA
    ~- [ ] Checked in both **light and dark** modes~
    ~- [ ] Checked for **accessibility** including keyboard-only and screenreader modes~
- Docs site QA - N/A
- Code quality checklist
- Release checklist
    ~- [ ] If applicable, added the **breaking change** issue label (and filled out the breaking change checklist)~
- Designer checklist - N/A
